pandas/core/arrays/base.py

```diff
@@ -781,17 +781,24 @@ def convert_values(param):
             # a TypeError should be raised
             res = [op(a, b) for (a, b) in zip(lvalues, rvalues)]
 
-            if coerce_to_dtype:
-                try:
-                    res = self._from_sequence(res)
-                except Exception:
+            def _maybe_convert(arr):
+                if coerce_to_dtype:
                     # https://github.com/pandas-dev/pandas/issues/22850
                     # We catch all regular exceptions here, and fall back
                     # to an ndarray.
-                    res = np.asarray(res)
+                    try:
+                        res = self._from_sequence(arr)
+                    except Exception:
+                        res = np.asarray(arr)
+                else:
+                    res = np.asarray(arr)
+                return res
+
+            if op.__name__ in {'divmod', 'rdivmod'}:
```

Comment:
@jbrockmendel can't we use dispatch_to_extension_op here to avoid duplication of code?

Reply:
I asked something similar a few days ago.  If Tom says it isn't feasible, I believe him.

Reply:
We are *inside* the extension array here, so it would also be strange to use (which doesn't prevent that both could share a helper function, if that would be appropriate). 
But here we need to construct the divmod correctly, while dispatch_to_extension_op should assume this is already done correctly by the EA

Reply:
> which doesn't prevent that both could share a helper function,

Right. This is possible, if people want it. I'll push up a commit with some kind of `do_extension_op` that both of these call to so people can take a look.

Reply:
Ah, now that I take a look it's not so straightforward. The two are similar but just slightly different in enough places that they wouldn't benefit from sharing code really.

1. The unboxing of values. `dispatch_to_extension_op` knows that at least one of the two is a Series[EA]. `_binop` knows that `self` is an EA.
2. The op: `dispatch_to_extension_op` dispatches, `_binop` is defining it in a list comprehension
3. The re-boxing: `_binop` has the whole maybe re-constructing `_from_seqence` that the `dispatch_to_extension_op` doesn't have to worry about at all.

Reply:
Yes, as said above, they are doing different things, so I agree it is not worth here.

Reply:
Which bits of code would be de-duplicated though (specific lines)? They three items I listed out are similar across the two, but different.

Reply:
> They do different things because this function is a bunch of stuff all put together

Sorry, but apart from that the above may be true, they still do fundamentally something different. The dispatch functions calls the EA to perform an operation, the above code *is* the EA doing the operation. There is nothing to be shared here.

I fully agree we should be careful not to create unnecessary technical debt, but that is IMO unrelated to this specific piece of code.

Reply:
> Which bits of code would be de-duplicated though (specific lines)? They three items I listed out are similar across the two, but different.

I would make an effort to try to fix this. It makes a future reader of this code wonder why there are 2 different implementations of basically the same thing. At the very least using something like

``def _construct_divmod_result(left, result, index, name, dtype=None):``

Its certainly easy to right *new* code. What is tricky is to integrate with our existing base. I don't think this goes far enough.

Reply:
Can you be more specific in how the existing `_construct_divmod_result` could be used in the above code?

Reply:
OK I don't understand what you want so feel free to push here.

But before you put too much time into it, keep in mind that at this point the version in `_binop` has a List of scalars, that *may* be convertible back to an ExtensionArray after unpacking into a tuple of lists. The version in `core/ops.py` already has a tuple of arrays., which it knows has to be converted to a Series.

Reply:
> OK I don't understand what you want so feel free to push here.

Because this is creating a very hard to understand situation where a future reader will not understand why we have 2 different places where handling ops. The current de-facto standard is pandas/core/ops.py

so now they have to understand / disambiguate why & where to see the code paths. These conceptually do exactly the same thing, so why 2. I don't know exactly what code should be change / integrated here. I would see if its is possible to integrate these rather than adding a bunch of new code.

Reply:
> I would see if its is possible to integrate these rather than adding a bunch of new code.

I started and in 
https://github.com/pandas-dev/pandas/pull/22932#discussion_r223328039 explained why that wasn't possible.

But if you have ideas that I don't see then feel free to push them here. But based on 

> I don't know exactly what code should be change / integrated here.

it sounds like you don't have anything specific in mind? If so, perhaps trust @jorisvandenbossche and I when we say that these are doing two different things, and so can't be shared?

Reply:
They do not do "conceptually exactly the same thing". Paraphrasing myself from above:

>  The dispatch function calls the EA to perform an operation, the above code *is* the EA doing the operation.

Why would those two different things necessarily need to live in the same place / code path?

Of course, we could still move the whole `EA._create_method` to `ops.py` (which would indeed be similar as functions like `add_flex_arithmetic_methods` in ops.py that is used in `series.py` to add methods to Series). But this is then not related to the change in this PR, and should be left for another issue/PR to discuss (personally I don't think that would be an improvement).

> I would see if its is possible to integrate these rather than adding a bunch of new code.

Well, and both Tom and me who have looked into the code, say: we don't think it is possible.

Reply:
@TomAugspurger I saw your comment. I also @jorisvandenbossche comments. I have not looked at this in detail, nor do I have time to. My point is that this instantly creates technical debt no matter how you slice it.

It may require some reorganization to integrate this, and I appreciate that. So happy to defer this, maybe @jbrockmendel has more insight.

Reply:
I'll be doing a sparse-de-duplication PR following #22880, can take a fresh look at this then.  In the interim, I wouldn't let this issue hold up this PR.

Reply:
> My point is that this instantly creates technical debt no matter how you slice it. 

It really doesn't. They're doing two different things.

Reply:
This PR is fixing a bug, not doing a refactor of how the ops on EA's are implemented\*\* . If somebody want to look into that, it should be done in a separate PR anyway. So merging.

\*\* and I fully acknowledge that sometimes, to properly fix a bug, you also need to refactor otherwise you just keep adding hacks. However, I don't think that is the case here, see all the comments above.